WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt

```diff
@@ -0,0 +1,363 @@
+package org.wordpress.android.ui.main
+
+import android.annotation.SuppressLint
+import android.app.Fragment
+import android.app.FragmentManager
+import android.app.FragmentTransaction.TRANSIT_FRAGMENT_FADE
+import android.content.Context
+import android.support.annotation.DrawableRes
+import android.support.annotation.IdRes
+import android.support.annotation.StringRes
+import android.support.design.internal.BottomNavigationItemView
+import android.support.design.internal.BottomNavigationMenuView
+import android.support.design.widget.BottomNavigationView
+import android.support.design.widget.BottomNavigationView.OnNavigationItemReselectedListener
+import android.support.design.widget.BottomNavigationView.OnNavigationItemSelectedListener
+import android.util.AttributeSet
+import android.util.SparseArray
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.animation.Animation
+import android.widget.ImageView
+import android.widget.TextView
+import org.wordpress.android.R
+import org.wordpress.android.ui.main.WPMainActivity.OnScrollToTopListener
+import org.wordpress.android.ui.notifications.NotificationsListFragment
+import org.wordpress.android.ui.prefs.AppPrefs
+import org.wordpress.android.ui.reader.ReaderPostListFragment
+import org.wordpress.android.util.AniUtils
+import org.wordpress.android.util.AniUtils.Duration
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+
+/*
+ * Bottom navigation view and related adapter used by the main activity for the
+ * four primary views - note that we ignore the built-in icons and labels and
+ * insert our own custom views so we have more control over their appearance
+ */
+class WPMainNavigationView : BottomNavigationView, OnNavigationItemSelectedListener,
+        OnNavigationItemReselectedListener {
+    private var mNavAdapter: NavAdapter? = null
+    private var mFragmentManager: FragmentManager? = null
+    private var mListener: OnPageListener? = null
+    private var mPrevPosition = -1
+
+    val activeFragment: Fragment?
+        get() = mNavAdapter!!.getFragment(currentPosition)
```

Comment:
try to avoid `!!` at all costs, in this case use `navAdapter?.getFragment` which doesn't crash if `navAdapter` is null